test/functional/interface_rest.py

```diff
@@ -151,23 +151,48 @@ def run_test(self):
         txid = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1)
         json_string = http_get_call(url.hostname, url.port, '/rest/tx/'+txid+self.FORMAT_SEPARATOR+"json")
         json_obj = json.loads(json_string)
-        vintx = json_obj['vin'][0]['txid'] # get the vin to later check for utxo (should be spent by then)
+        # get the spent output to later check for utxo (should be spent by then)
+        spent = '{}-{}'.format(json_obj['vin'][0]['txid'], json_obj['vin'][0]['vout'])
         # get n of 0.1 outpoint
         n = 0
         for vout in json_obj['vout']:
             if vout['value'] == 0.1:
                 n = vout['n']
+        spending = '{}-{}'.format(txid, n)
 
-        json_request = '/'+txid+'-'+str(n)
+        json_request = '/'+spending
         json_string = http_get_call(url.hostname, url.port, '/rest/getutxos'+json_request+self.FORMAT_SEPARATOR+'json')
         json_obj = json.loads(json_string)
-        assert_equal(len(json_obj['utxos']), 0) #there should be an outpoint because it has just added to the mempool
+        assert_equal(len(json_obj['utxos']), 0) #there should be no outpoint because it has just added to the mempool
```

Comment:
I think this comment could be clearer. Something like `# tx not found since it is in the mempool (and /mempool not included in request`

edit: in fact, I think you could remove all the inline comments and just place a comment at the top of the section explaining what's happening (create a transaction, check that it's found with `/checkmempool`, but not found without, then confirm the transaction and check that it's found with or without `/checkmempool`

Reply:
Thanks!
I agree, and would be happy to fix this issue (together with a general tidying-up of `interface_rest.py` tests) in a following PR.